ddtrace/tracer/span.go

```diff
@@ -530,10 +530,10 @@ func (s *Span) setMetric(key string, v float64) {
 		if v == float64(samplernames.AppSec) {
 			s.setSamplingPriorityLocked(ext.PriorityUserKeep, samplernames.AppSec)
 		}
-	case ext.SamplingPriority:
-		// ext.SamplingPriority is deprecated in favor of ext.ManualKeep and ext.ManualDrop.
-		// We have it here for backward compatibility.
-		s.setSamplingPriorityLocked(int(v), samplernames.Manual)
+	case ext.ManualDrop:
+		if v == float64(samplernames.AppSec) {
```

Comment:
I am not sure we need to add this here to be honest. There is a weird use case for ManualKeep, defined by appsec rules. But they don't use the same convention for manualDrop 🤔

Reply:
@Julio-Guerra Any idea why this there?

Reply:
Did you git-blame the file? It wasn't me who wrote those lines but @Barbayar a few years ago when he worked on distributed tracing on all tracers. IIRC, it was just about the decision maker tag `_dd.p.dm`.

ASM doesn't rely on those details, we're good as long as we can force-keep the traces when we have security events.
The only ASM RFC where we went into such details is the standalone ASM billing [RFC](https://docs.google.com/document/d/12NBx-nD-IoQEMiCRnJXneq4Be7cbtSc6pJLOFUWTpNE/edit) and they are system-tested so regressions would be found if any.

Reply:
Reading the PR description, I believe it would be great to keep a lower-level API for internal usage that provides more options. But likely bonus as far as I am aware of our product needs.

Reply:
Yes, I spoke with @Barbayar and it seems that AppSec has a few cases where we want to keep specific traces, which is why I decided to keep the case. I'll keep the `ManualKeep` case, but seems like we don't need the `ManualDrop`, so I've removed it. Thanks!